usr/src/lib/brand/lx/zone/lx_boot_zone_ubuntu.ksh

```diff
@@ -28,30 +29,32 @@ safe_dir /etc/resolvconf/resolv.conf.d
 safe_dir /etc/network
 safe_dir /etc/network/interfaces.d
 safe_dir /etc/network/interfaces.d/smartos
-
-# Populate resolve.conf setup files
-zonecfg -z $ZONENAME info attr name=resolvers | awk '
-BEGIN {
-	print("# AUTOMATIC ZONE CONFIG")
-}
-$1 == "value:" {
-	nres = split($2, resolvers, ",");
-	for (i = 1; i <= nres; i++) {
-		print("nameserver", resolvers[i]);
-	}
-}
-' > $tmpfile
-zonecfg -z $ZONENAME info attr name=dns-domain | awk '
-$1 == "value:" {
-	dom = $2
-}
-END {
-	print("search", dom);
-}
-' >> $tmpfile
-fnm=$ZONEROOT/etc/resolvconf/resolv.conf.d/tail
-if [[ -f $fnm || -h $fnm || ! -e $fnm ]]; then
-	mv -f $tmpfile $fnm
+safe_dir /etc/systemd
+
+# Populate resolv.conf setup files IFF we have resolvers information.
+resolvers=`zone_attr resolvers`
+if [[ $? == 0 ]]; then
+
+    echo "# AUTOMATIC ZONE CONFIG" > $tmpfile
+    _IFS=$IFS; IFS=,; for r in $resolvers; do
+        echo "nameserver $r"
+    done >> $tmpfile
+    IFS=$_IFS
+    domain=`zone_attr dns-domain`
+    [[ $? == 0 ]] && echo "search $domain" >> $tmpfile
+
+    if [ -f $ZONEROOT/etc/systemd/resolved.conf ]; then
+        cf=$ZONEROOT/etc/systemd/resolved.conf
+	sed -i -E '/^(DNS|Domains) *=/d' $cf
+        echo "DNS=$resolvers" >> $cf
+        [[ -n "$domain" ]] && echo "Domains=$domain" >> $cf
+        mv -f $tmpfile $ZONEROOT/etc/resolv.conf
```

Comment:
@papertigers pointed out to me that this will overwrite resolv.conf REGARDLESS of whether or not there's a file there already.

Reply:
That's how it work in OmniOS, for better or worse. If you define the resolvers and domain in the zone config, then it's regenerated at boot. It might not be right for SmartOS.

Reply:
OTOH we appear to do this already for other LX zones (see the else case below...).

Reply:
on SmartOS, we have the `maintain_resolvers` field in vmadm, which OS zones only update resolvers if that's set.

This would present a change to users. It would mean that they can't overwrite the resolvers, or need to do it on every boot.

Reply:
Is it a regression from current LX behavior, though?  The old code suggests maybe it isn't a regression and it's always been different vs. OS zones?

Reply:
If this is the current LX behavior then we're good.